float64  foo  # 2.17
int32[3] bar  # [8, 2, 5]